internal/xds/balancer/clusterresolver/e2e_test/dns_impl_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2025 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/internal/stubserver"
+	"google.golang.org/grpc/internal/testutils/xds/e2e"
+	"google.golang.org/grpc/internal/xds/bootstrap"
+	"google.golang.org/grpc/internal/xds/xdsclient"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/resolver/manual"
+	"google.golang.org/grpc/serviceconfig"
+
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+
+	_ "google.golang.org/grpc/internal/xds/balancer/clusterresolver"
+)
+
+// TestLogicalDNS_MultipleEndpoints tests the cluster_resolver LB policy
+// using a LOGICAL_DNS discovery mechanism.
+//
+// The test verifies that multiple addresses returned by the DNS resolver are
+// grouped into a single endpoint (as per gRFC A61). Because the round_robin
+// LB policy (configured via xdsLbPolicy) sees only one endpoint, it should
+// not rotate traffic between the addresses. Instead, the single endpoint
+// (which contains all addresses) is picked, and connects to the first address.
+func (s) TestLogicalDNS_MultipleEndpoints(t *testing.T) {
+	// Spin up a management server to receive xDS resources from.
+	managementServer := e2e.StartManagementServer(t, e2e.ManagementServerOptions{})
+
+	// Create bootstrap configuration pointing to the above management server.
+	nodeID := uuid.New().String()
+	bootstrapContents := e2e.DefaultBootstrapContents(t, nodeID, managementServer.Address)
+
+	// Start backend servers which provide an implementation of the TestService.
+	server1 := stubserver.StartTestService(t, nil)
+	defer server1.Stop()
+	server2 := stubserver.StartTestService(t, nil)
+	defer server2.Stop()
+
+	// Mock the DNS Resolver
+	const dnsScheme = "dns"
+	dnsR := manual.NewBuilderWithScheme(dnsScheme)
```

Comment:
Does this not interfere with the DNS resolver used for the grpc channel between the xDS client and the management server?